cpp/src/arrow/python/helpers.h

```diff
@@ -55,7 +54,10 @@ PyObject* DecimalFromString(PyObject* decimal_constructor,
                             const std::string& decimal_string);
 Status DecimalFromPythonDecimal(PyObject* python_decimal, const DecimalType& arrow_type,
                                 Decimal128* out);
-bool IsPyInteger(PyObject* obj);
+bool IsPyInteger(const PyObject* obj);
```

Comment:
Why the consts? I don't think it makes sense, and you're bound to do a lot of casts as soon as you call the Python C API.

Reply:
Note `PyObject` is non-const pretty much by construction, as it has a reference count that can be mutated by any operation.

Reply:
Ok, I was following convention in the file. I can adjust.

Reply:
I've been removing all the instances of `const PyObject*` after having a compiler warning from an internal Py-API, so happy to see them all go